src/as-cache.c

```diff
@@ -829,6 +832,16 @@ as_cache_open (AsCache *cache, const gchar *fname, const gchar *locale, GError *
 		goto fail;
 	}
 
+	/* set FD_CLOEXEC manually. LMDB should do that, but it doesn't:
+	   https://www.openldap.org/lists/openldap-bugs/201702/msg00003.html */
+	rc = mdb_env_get_fd (priv->db_env, &db_fd);
+	if (rc == MDB_SUCCESS) {
+		db_fd_flags = fcntl (db_fd, F_GETFD);
```

Comment:
Could you move the `db_fd_flags` variable into this block, so it only is visible when we actually need it, and also move the `if (db_fd_flags != -1)` into this block as well, to logically group this together?

Reply:
It looked to me like it was using the old C style to declare all variables at the function start...

Done.

Reply:
Yeah, some older parts of the code do, but for new code we can use fancy "new" things! - Even the `cleanup` attribute and `g_autofree`, which is one of the best things that ever happened to a C developer ;-)